hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/InputInfo.java

```diff
@@ -0,0 +1,352 @@
+       *
+       * @param clientInfo
+       *          Accumulo connection information
+       */
+      TableParams clientInfo(ClientInfo clientInfo);
+
+      /**
+       * Set the connection information needed to communicate with Accumulo in this job.
+       *
+       * @param clientProps
+       *          Accumulo connection information
+       */
+      TableParams clientProperties(Properties clientProps);
+    }
+
+    interface TableParams {
+      /**
+       * Sets the name of the input table, over which this job will scan.
+       *
+       * @param tableName
+       *          the table to use when the tablename is null in the write call
+       */
+      AuthsParams table(String tableName);
+    }
+
+    interface AuthsParams {
+      /**
+       * Sets the {@link Authorizations} used to scan. Must be a subset of the user's
+       * authorizations. If none present use {@link Authorizations#EMPTY}
+       *
+       * @param auths
+       *          the user's authorizations
+       */
+      InputFormatOptions scanAuths(Authorizations auths);
+    }
+
+    interface BatchScanOptions {
+      /**
+       * @return newly created {@link InputInfo}
+       */
+      InputInfo build();
+    }
+
+    interface NonBatchScanOptions {
+      /**
+       * @see InputFormatOptions#scanIsolation()
+       */
+      NonBatchScanOptions scanIsolation();
+
+      /**
+       * @see InputFormatOptions#localIterators()
+       */
+      NonBatchScanOptions localIterators();
+
+      /**
+       * @see InputFormatOptions#offlineScan()
+       */
+      NonBatchScanOptions offlineScan();
+
+      /**
+       * @return newly created {@link InputInfo}
+       */
+      InputInfo build();
+
+    }
+
+    /**
+     * Optional values to set using fluent API
+     */
+    interface InputFormatOptions {
+      /**
+       * Sets the name of the classloader context on this scanner
+       *
+       * @param context
+       *          name of the classloader context
+       */
+      InputFormatOptions classLoaderContext(String context);
+
+      /**
+       * Sets the input ranges to scan for the single input table associated with this job.
+       *
+       * @param ranges
+       *          the ranges that will be mapped over
+       * @see TableOperations#splitRangeByTablets(String, Range, int)
+       */
+      InputFormatOptions ranges(Collection<Range> ranges);
+
+      /**
+       * Restricts the columns that will be mapped over for this job for the default input table.
+       *
+       * @param cfcqPairs
+       *          a pair of byte[] objects corresponding to column family and column qualifier. If
+       *          the column qualifier is null, the entire column family is selected. An empty set
+       *          is the default and is equivalent to scanning the all columns.
+       */
+      InputFormatOptions fetchColumns(Collection<Pair<byte[],byte[]>> cfcqPairs);
```

Comment:
What do you think of changing this to just "fetchColumn(byte[] colFam, byte[] colQual)" like the one we have in ScannerBase but byte[] instead of Text?  Or we could use IteratorSetting.Column class from o.a.a.core.client and have just one parameter.

Reply:
Fixed in 3fa7b4472877a713748e9bbd1ca3e89886f4cb7a